internal/iam/role.go

```diff
@@ -327,3 +328,34 @@ func (p *projectRole) Actions() map[string]action.Type {
 	ret[action.SetPrincipals.String()] = action.SetPrincipals
 	return ret
 }
+
+// getRoleScopeId returns scopeID for the Role from the base type iam_role table
```

Comment:
Since this uses a reader, I would place this in one of the `repository_` files. It should really only be called by a repository, so it would make it easier to find it if is in the same file as the repository.